src/main/java/com/rtbhouse/utils/avro/FastDeserializerGenerator.java

```diff
@@ -267,11 +290,151 @@ private void processRecord(JVar schemaVariable, JVar containerVariable, final Sc
             }
         }
 
+        // Handle default values
+        if (recordAction.getShouldRead()) {
+            Set<String> fieldNamesSet = record.getFields().stream().map(Schema.Field::name).collect(Collectors.toSet());
+            for (Schema.Field readerField : readerRecord.getFields()) {
+                if (!fieldNamesSet.contains(readerField.name())) {
+                    boolean success = seekToDefault(actionIterator);
+                    if (!success || readerField.defaultValue() == null) {
+                        throw new FastDeserializerGeneratorException(
+                                "No default value defined for new field: " + readerField.name());
+                    }
+
+                    seekFieldAction(true, readerField, actionIterator);
+                    JVar schemaVar = null;
+                    if (useGenericTypes) {
+                        schemaVar = declareSchemaVariableForRecordField(readerField.name(), readerField.schema(),
+                                schemaVariable);
+                    }
+                    JExpression value = parseDefaultValue(readerField.schema(), readerField.defaultValue(), body,
+                            schemaVar, readerField.name());
+                    body.invoke(result, "put").arg(JExpr.lit(readerField.pos())).arg(value);
+                }
+            }
+        }
+
         if (recordAction.getShouldRead()) {
             body._return(result);
         }
     }
 
+    private JExpression parseDefaultValue(Schema schema, JsonNode defaultValue, JBlock body, JVar schemaVariable,
+            String fieldName) {
+        Schema.Type schemaType = schema.getType();
+
+        // Default value of union have to have first type from union
```

Comment:
I would change the comment to something like: "The default value of union is the first defined type"